### ISSUE TITLE
Cannot switch back to theme

### ISSUE THREAD

**Issue by nicos68** [svrana/neosolarized.nvim#30]
If nvim is set to any other color theme, then calling `:colorscheme neosolarized` has no effect. Should I use another command(s) if I want to switch back to neosolarized ?

**Comment by tan-wei** [comment]
Met the same issue. And I can confirm that if we call `:colorscheme neosolarized` and then query which colorscheme we used, it just prints "default".

**Comment by svrana** [comment]
While it'd be easy enough to make `:colorscheme neosolarized` switch to neosolarized (by calling lua require('neosolarized').setup() in colors/neosolarized.vim) it would not preserve any options you passed into setup the first time. In retrospect, this makes the use of a setup function for a colorscheme not the greatest fit. I'm hesitant to add this since it won't always work, but maybe it's better than nothing.

**Comment by nicos68** [comment]
Actually to be fair I decided to move away from this color scheme. I stayed with it for a while because I find it very satisfying visually but I recently changed office and needed to switch between light and dark theme more often because of lighting conditions. I started fiddling with the `setup` method and was making progress but I still had bugs and problems when switching back and forth between themes like nvim tree staying with the previous theme, or broken colors in the completion popup. So I gave up and decided it was easier using more advanced themes that handle all those cases like [nightfox ](https://github.com/EdenEast/nightfox.nvim) or [catppuccin](https://github.com/catppuccin/nvim).

Still a great theme if you don’t need to switch though. Maybe retrofitting it with tools like [lush](https://github.com/rktjmp/lush.nvim) or [themer](https://github.com/themercorp/themer.lua) could bing improvements …
